File: backend/src/xfd_django/xfd_api/schema_models/role.py
"""Role schema."""
# Standard Python Libraries
from datetime import datetime
from typing import Any, Optional
from uuid import UUID

# Third-Party Libraries
from pydantic import BaseModel


class Role(BaseModel):
    """Role schema."""

    id: UUID
    created_at: datetime
    updated_at: datetime
    role: str
    approved: bool
    created_by_id: Optional[Any]
    approved_by_id: Optional[Any]
    user: Optional[Any]
    organization_id: Optional[Any]

    class Config:
        """Config."""

        from_attributes = True

File: frontend/src/test-utils/role.ts
import { testOrganization } from './organization';
import { testUser } from './user';

export const testRole = {
  id: 'role-1',
  created_at: '2024-01-01T00:00:00Z',
  updated_at: '2024-01-01T00:00:00Z',
  role: 'user' as 'user' | 'admin',
  user: testUser,
  organization: testOrganization,
  approved: true
};
